src/node_file.cc

```diff
@@ -532,16 +533,17 @@ inline FSReqBase* AsyncDestCall(Environment* env,
     uv_fs_t* uv_req = req_wrap->req();
     uv_req->result = err;
     uv_req->path = nullptr;
-    after(uv_req);
+    after(uv_req);  // after may delete req_wrap if there is an error
     req_wrap = nullptr;
   }
 
   if (req_wrap != nullptr) {
```

Comment:
Since you're here, maybe change this `if` to an `else` of the previous conditional.  Makes the logic a little more obvious, IMO.